Source/Particles/MultiParticleContainer.cpp

```diff
@@ -484,3 +492,65 @@ MultiParticleContainer::doContinuousInjection() const
     }
     return warpx_do_continuous_injection;
 }
+
+/* \brief Get ID of target species of each species.
+ * The users specifies the name of the target species, 
+ * this routine get its ID.
+ */
+void
+MultiParticleContainer::mapSpeciesTarget()
+{
+    for (int i=0; i<nspecies; i++){
+        auto& pc = allcontainers[i];
+        // If species pc has ionization on, find species with name 
+        // pc->ionization_target_name and store its ID into 
+        // pc->ionization_target.
+        if (pc->do_field_ionization){
+            int i_target = getSpeciesID(pc->ionization_target_name);
+            pc->ionization_target = i_target;
+        }
+    }
+}
+
+/* \brief Given a species name, return its ID.
+ */
+int
+MultiParticleContainer::getSpeciesID(std::string target_str)
+{
+    int i_target;
+    bool found = 0;
+    // Loop over species
+    for (int i=0; i<nspecies; i++){
+        // If species name matches, store its ID
+        // into i_target
+        if (species_names[i] == target_str){
+            found = 1;
+            i_target = i;
+        }
+    }
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(found != 0,
+        "ERROR: could not find ID for species");
+    return i_target;
+}
+
+/* \brief Find ionized particles and create target electrons
+ */
+void
+MultiParticleContainer::doFieldIonization()
+{
+    for (auto& pc : allcontainers){
+        for (int lev = 0; lev <= pc->finestLevel(); ++lev){
+            // If ionization off for this species, do not do anything.
+            if (!pc->do_field_ionization){ continue; }
+            // Fill MultiParticleContainer::pc_tmp with a copy of all
+            // ionizable particles.
+            pc->copyParticles(lev);
```

Comment:
This calls a member function of `PhysicalParticleContainer`, that modifies `MultiParticleContainer::pc_tmp`. Is that OK?